### PR TITLE
feature: refactored reports module for Vue application

### LINKED ISSUE #232: Milestone 1.1: Remove HTML code fragments from reports module
Remove HTML fragments from reports module in preparation for the interactive Vue report

### PR DESCRIPTION
**Pull request by sankha555**
* [x] partially fixes #232 
* [x] removed HTML code fragments from reports module to make it compatible for JSON serialization for the Vue application

### CODE REVIEW

**Comment by sankha555** [comment]
@matthiaskoenig I have refactored the following files to replace the HTML fragments with the dictionary representation of the objects:
- ```sbmlinfo.py```
- ```formating.py```

Tests might fail for now due the changes. Please take a look at the updates and let me know what improvements should  be made. I can then add some appropriate tests too to unit test the new format. Thanks!

File: src/sbmlutils/report/sbmlinfo.py [hunk 1]
```diff
@@ -610,28 +587,29 @@ def info_events(self) -> List[Dict[str, Any]]:
...
+                "persistent": trigger.persistent
+            }
+
+            priority = ""
             if event.isSetPriority():
                 priority = event.getPriority()
             info["priority"] = priority
 
-            delay = empty_html()
+            delay = ""
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/sbmlinfo.py:601]: This should not be empty strings, but None !.
Also use the following pattern consistently for attributes
```
info["delay"] = event.getDelay() if event.isSetDelay() else None

```

File: src/sbmlutils/report/sbmlinfo.py [hunk 2]
```diff
@@ -610,28 +587,29 @@ def info_events(self) -> List[Dict[str, Any]]:
...
-                """
-
-            priority = empty_html()
+            info["trigger"] = {
+                "math": math(trigger, self.math_render),
+                "initial_value": trigger.initial_value,
+                "persistent": trigger.persistent
+            }
+
+            priority = ""
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/sbmlinfo.py:596]:
```
info["priority"] = event.getPriority() if event.isSetPriority() else None
```

File: src/sbmlutils/report/formating.py [hunk 3]
```diff
@@ -7,6 +7,8 @@
 from sbmlutils.metadata import miriam
 from sbmlutils.report import mathml
 
+from typing import Dict, List
+
 
 def annotation_to_html(item: libsbml.SBase) -> str:
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:13]: There should be no `*_to_html` method any more ?

File: src/sbmlutils/report/formating.py [hunk 4]
```diff
@@ -36,6 +38,37 @@ def annotation_to_html(item: libsbml.SBase) -> str:
...
+
+    :param item: SBase instance
+    """
+    res = {}
+    res["cvterms"] = []
+    for kcv in range(item.getNumCVTerms()):
+        info = {}
+        cv = item.getCVTerm(kcv)
+        q_type = cv.getQualifierType()
+        if q_type == 0:
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:52]: check against correct type code from libsbml not int

File: src/sbmlutils/report/formating.py [hunk 5]
```diff
@@ -36,6 +38,37 @@ def annotation_to_html(item: libsbml.SBase) -> str:
...
+            qualifier = miriam.ModelQualifierType[cv.getModelQualifierType()]
+        elif q_type == 1:
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:54]: check against correct type code

File: src/sbmlutils/report/formating.py [hunk 6]
```diff
@@ -206,30 +237,32 @@ def modelHistoryToString(mhistory: libsbml.ModelHistory) -> str:
...
-    items.append("<b>Creator</b>")
+        return {}
+    creators = []
     for kc in range(mhistory.getNumCreators()):
-        cdata = []
+        cdata = {}
         c = mhistory.getCreator(kc)
         if c.isSetGivenName():
-            cdata.append(c.getGivenName())
+            cdata["given_namw"] = c.getGivenName()
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:246]: typo

File: src/sbmlutils/report/formating.py [hunk 7]
```diff
@@ -206,30 +237,32 @@ def modelHistoryToString(mhistory: libsbml.ModelHistory) -> str:
...
-        return ""
-    items = []
-    items.append("<b>Creator</b>")
+        return {}
+    creators = []
     for kc in range(mhistory.getNumCreators()):
-        cdata = []
+        cdata = {}
         c = mhistory.getCreator(kc)
         if c.isSetGivenName():
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:245]: use pattern
```
cdata["given_name"] = c.getGivenName() if c.isSetGivenName() else None
```
I.e. use oneliners and set None as default so the fields exist.

File: src/sbmlutils/report/formating.py [hunk 8]
```diff
@@ -358,50 +468,60 @@ def notes(item: libsbml.SBase) -> str:
...
 
 
-def cvterm(item: libsbml.SBase) -> str:
+def cvterm(item: libsbml.SBase) -> Dict:
     """Create HTML code fragment enclosing cvterm data for the item.
 
     :param item: SBML object for which cvterm data has to be displayed
     :return: HTML code fragment enclosing cvterm data for the item
     """
     if item.isSetAnnotation():
```
Comment (matthiaskoenig) [inline src/sbmlutils/report/formating.py:477]: Makes more sense to have this check in `annotation_dict`

**Review by matthiaskoenig — COMMENTED** [review]
Most of the things are okay. This is difficult to review and we have to probably make a code review together on this. I suggest you finish this up as well as the JSON serialization and then we would refactor the code together in a session.

**Comment by sankha555** [comment]
@matthiaskoenig Sure Sir, I will resolve the suggested reviews first and then move to the JSON serialization. We can then have a session on reviewing the code changes, and then I can move on to the next module. Thanks for the reviews!

**Comment by sankha555** [comment]
@matthiaskoenig I have added the suggested changes and the serialization. I have also added a small test to print the json to a file, but unfortunately it is not returning any results. I had tried to print dummy results as well but the functions in ```sbmlreport.py``` do not seem to be returning any value (everytime ```None``` is returned, even if I try to return a simple string like "Hello"). 

Could you please take a look at the changes and also suggest what might be perhaps causing the above issue? Please let me know if you need some more information from me. I can also be available for reviewing the code with you at your time and convenience. Thanks!

**Pull request merged**